pkgs/by-name/st/stackql/package.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
+, testers
+, stackql
+}:
+
+buildGoModule rec {
+  pname = "stackql";
+  version = "0.5.643";
+
+  src = fetchFromGitHub {
+    owner = "stackql";
+    repo = "stackql";
+    rev = "v${version}";
+    hash = "sha256-9W6bEI+5Q0Kgbd14sWKde3I6WIVz1ZxsXmR009mOPog=";
+  };
+
+  vendorHash = "sha256-xcly4jtdUkx/s+YWYFBVeuI2kQBu2oqbLN9ZKkHppkA=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/stackql/stackql/internal/stackql/cmd.BuildMajorVersion=0"
+    "-X github.com/stackql/stackql/internal/stackql/cmd.BuildMinorVersion=5"
+    "-X github.com/stackql/stackql/internal/stackql/cmd.BuildPatchVersion=643"
+    "-X github.com/stackql/stackql/internal/stackql/cmd.BuildCommitSHA=${src.rev}"
+    "-X github.com/stackql/stackql/internal/stackql/cmd.BuildShortCommitSHA=${src.rev}"
+    "-X github.com/stackql/stackql/internal/stackql/cmd.BuildDate=2024-05-15T07:51:52Z" # date of commit hash
+    "-X stackql/internal/stackql/planbuilder.PlanCacheEnabled=true"
+  ];
+
+  # Don't run checks because of the following failure in sandbox
+  # --- FAIL: TestSelectComputeDisksOrderByCrtTmstpAsc (0.01s)
+  # panic: httptest: failed to listen on a port: listen tcp6 [::1]:0: bind: operation not permitted
```

Comment:
do any tests pass ?

Reply:
tests now passing.

Reply:
then we don't need `doCheck = false` ?

Reply:
testVersion is now passing.

```
Running phase: checkPhase
ok      github.com/stackql/stackql/internal/stackql/acid/binlog 0.341s
ok      github.com/stackql/stackql/internal/stackql/acid/txn_context    0.256s
ok      github.com/stackql/stackql/internal/stackql/astanalysis/selectmetadata  0.354s
ok      github.com/stackql/stackql/internal/stackql/datasource/sql_datasource   0.332s
ok      github.com/stackql/stackql/internal/stackql/datasource/sqltable 0.354s
--- FAIL: TestSelectComputeDisksOrderByCrtTmstpAsc (0.02s)
panic: httptest: failed to listen on a port: listen tcp6 [::1]:0: bind: operation not permitted [recovered]
        panic: httptest: failed to listen on a port: listen tcp6 [::1]:0: bind: operation not permitted
```

Some other tests pass, but the test that listens on a port fails as above.

Reply:
you can add 

```
  # -run and -skip accept regular expressions
  checkFlags = [
    "-skip=^TestSelectComputeDisksOrderByCrtTmstpAsc$"
  ];
```

just to skip failing tests

as per https://nixos.org/manual/nixpkgs/unstable/#sec-language-go

Reply:
skipped all known network tests

Reply:
all tests pass when using same tags as used by upstream

```
checkFlags = [ "--tags json1,sqleanal" ];
```